ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -358,27 +360,323 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
+
+            GlobalDimMatrixType const K_over_mu = K / mu;
+
+            const double drho_dp = _process_data.fluid_properties->getdValue(
+                MaterialLib::Fluid::FluidPropertyType::Density,
+                vars,
+                MaterialLib::Fluid::PropertyVariableType::p);
+            const double drho_dC = _process_data.fluid_properties->getdValue(
+                MaterialLib::Fluid::FluidPropertyType::Density,
+                vars,
+                MaterialLib::Fluid::PropertyVariableType::C);
+
+            // matrix assembly
+            local_M.noalias() += w * N.transpose() * porosity * drho_dp * N;
+            local_K.noalias() +=
+                w * dNdx.transpose() * density * K_over_mu * dNdx;
+
+            if (_process_data.has_gravity)
+                local_b.noalias() +=
+                    w * density * density * dNdx.transpose() * K_over_mu * b;
+
+            // coupling term
+            {
+                double C0_int_pt = 0.0;
+                NumLib::shapeFunctionInterpolate(local_C0, N, C0_int_pt);
+
+                local_b.noalias() -= w * N.transpose() * porosity * drho_dC *
+                                     (C_int_pt - C0_int_pt) / dt;
+            }
+        }
+    }
+
+    void assembleComponentTransportEquation(
+        double const t, std::vector<double>& local_M_data,
+        std::vector<double>& local_K_data,
+        std::vector<double>& /*local_b_data*/,
+        LocalCoupledSolutions const& coupled_xs)
+    {
+        auto const transport_process_id = coupled_xs.process_id;
+        auto local_C = Eigen::Map<const NodalVectorType>(
+            coupled_xs.local_coupled_xs[transport_process_id].data(),
+            concentration_size);
+        auto local_p = Eigen::Map<const NodalVectorType>(
+            coupled_xs.local_coupled_xs[hydraulic_process_id].data(),
+            pressure_size);
+        auto local_p0 = Eigen::Map<const NodalVectorType>(
+            coupled_xs.local_coupled_xs0[hydraulic_process_id].data(),
+            pressure_size);
+
+        auto const dt = coupled_xs.dt;
+
+        auto local_M = MathLib::createZeroedMatrix<LocalBlockMatrixType>(
+            local_M_data, concentration_size, concentration_size);
+        auto local_K = MathLib::createZeroedMatrix<LocalBlockMatrixType>(
+            local_K_data, concentration_size, concentration_size);
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        SpatialPosition pos;
+        pos.setElementID(_element.getID());
+
+        auto const& b = _process_data.specific_body_force;
+
+        MaterialLib::Fluid::FluidProperty::ArrayType vars;
+
+        GlobalDimMatrixType const& I(
+            GlobalDimMatrixType::Identity(GlobalDim, GlobalDim));
+
+        auto const& transport_properties = _process_data.getTransportProperties(
+            _element.getID(),
+            _process_variables[transport_process_id][0].get().getName());
```

Comment:
I'd think this `[0]` is a component id here.

Reply:
This [0] is a product of using staggered coupling scheme. Component id is the transport process id.

Reply:
I think the intention of the code would be more clear if you would introduce somewhere above
```auto const& transport_process_variables = _process_variables[transport_process_id]```
or even
```auto const& first_transport_process_variable = _process_variables[transport_process_id][0]```.

Reply:
Fixed.